### ISSUE TITLE
fitscheck should not be using astropy.tests.helper.catch_warnings

### ISSUE THREAD

**Issue by pllim** [astropy/astropy#10528]
### Description

`fitscheck` uses `astropy.tests.helper.catch_warnings`:

https://github.com/astropy/astropy/blob/49708febbdaa1d1bf53f96981bf1f1d71295d33d/astropy/io/fits/scripts/fitscheck.py#L134

It should not be using a function from "test helper" module because that makes `pytest` a compulsory dependency whenever this import happens:

https://github.com/astropy/astropy/blob/49708febbdaa1d1bf53f96981bf1f1d71295d33d/astropy/io/fits/scripts/fitscheck.py#L47

### Expected behavior

Any non-test function should not be using stuff in `astropy.tests`.

### Blocker

An attempt to replace it with `warnings.catch_warnings` failed in #10504 , which @saimn explained beautifully at https://github.com/astropy/sphinx-astropy/issues/21#issuecomment-653692416 : 

*Basically Astropy's `catch_warnings` is overwriting the warnings defined in `setup.cfg` for pytest, by calling `treat_deprecations_as_exceptions` in its `__exit__` method. So the global `error` clause in pytest's config was ignored once Astropy's `catch_warnings` was called... When using `warnings.catch_warnings` this does not happen, so the global `error` clause is there, and the checksum verification in `io.fits` is issuing warning when the checksum is wrong, which turns into errors, which break the code's logic...*
